tests/playwright/sanity/assets/js/frontend/utils/url-actions.test.js

```diff
@@ -0,0 +1,184 @@
+		await editor.addElement( {
+			elType: 'widget',
+			widgetType: 'image-gallery',
+			settings: {
+				gallery_link: 'file',
+				open_lightbox: 'yes',
+			},
+		} );
+
+		await page.click( '.elementor-control-gallery-add' );
+		await page.click( 'text=Media Library' );
+		await page.waitForTimeout( 1000 );
+		await page.waitForSelector( 'li[tabindex="0"]' );
+
+		// Check if field image is already uploaded.
+		const fieldImage = await page.$( '[aria-label="field-image"]' );
+
+		if ( fieldImage ) {
+			await page.click( '[aria-label*="field-image"]' );
+		} else {
+			// Upload and select the field image
+			await page.click( 'text=Upload files' );
+			await page.waitForSelector( 'text=Drop files to upload' );
+			await page.setInputFiles( 'input[type="file"]', './tests/playwright/resources/field-image.jpg' );
+			// If this step is successful, the image should be already selected for the gallery.
+			await page.waitForSelector( 'text=field-image.jpg' );
+		}
+
+		// Select the mountain image to add to the gallery.
+		await page.click( '[aria-label="mountain-image"]' );
+
+		// Create the gallery.
+		await page.click( wpMediaAddButtonSelector );
+
+		// Could take time to create the gallery.
+		await page.waitForTimeout( 500 );
+
+		// Insert the gallery.
+		await page.click( 'text=Insert gallery', { delay: 500, clickCount: 2 } );
+
+		await page.waitForTimeout( 500 );
+
+		const galleryItems = editor.getPreviewFrame().locator( '.elementor-image-gallery figure.gallery-item' );
+
+		// Check that a gallery with two images has been added to the page.
+		await expect( galleryItems ).toHaveCount( 2 );
+
+		// Open the gallery's slideshow in a lightbox.
+		galleryItems.first().click();
+
+		const slideshowLightboxImage = editor.getPreviewFrame().locator( 'img[alt="mountain-image"].elementor-lightbox-image, img[alt="field-image"].elementor-lightbox-image' );
+
+		// Test that the lightbox appeared.
+		await expect( slideshowLightboxImage.first() ).toBeVisible();
+
+		// Save the page so we can run the front end tests.
+		await page.evaluate( () => $e.run( 'document/save/default' ) );
+
+		const frontendSlug = '/?p=' + editor.postId;
+
+		// Go to the front end of the test page.
+		await page.goto( frontendSlug );
+
+		/**
+		 * Get the action hash from the image, go to the page URL with the hash and check that the lightbox is triggered.
+		 */
+		const frontendMountainImageElement = await page.$( '.elementor-widget-image a[href*="mountain-image"]' );
+
+		const mountainImageHash = await frontendMountainImageElement.getAttribute( 'e-action-hash' );
+
+		const pageURL = page.url();
+
+		// Go to the front end of the test page with the action hash.
+		await page.goto( pageURL + mountainImageHash );
+
+		// Reload so the hash will be parsed.
+		await page.reload();
+
+		await page.waitForSelector( '.elementor-lightbox-image' );
+
+		const frontendSinglelightboxImage = page.locator( '.elementor-lightbox-image' );
```

Comment:
Please open the lightbox on the front and ensure it works

Reply:
@matipojo The test does that